appservice/src/intTest/groovy/nu/yona/server/VpnConnectionStatusEventTest.groovy

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Stichting Yona Foundation
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v.2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server
+
+import static nu.yona.server.test.CommonAssertions.*
+
+import groovy.json.*
+import nu.yona.server.test.User
+
+class VpnConnectionStatusEventTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Richard posts initial VPN connected event: connection status updated and buddies informed'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		when:
+		def response = richard.devices[0].postVpnStatus(appService, true)
+
+		then:
+		assertResponseStatusNoContent(response)
+
+		User richardAfterReload = appService.reloadUser(richard)
+		User bobAfterReload = appService.reloadUser(bob)
+		!richard.devices[0].vpnConnected
+		richardAfterReload.devices[0].vpnConnected
+		!bob.buddies[0].user.devices[0].vpnConnected
+		bobAfterReload.buddies[0].user.devices[0].vpnConnected
+
+		def bobMessagesAfterUpdate = appService.getMessages(bob)
+		assertResponseStatusOk(bobMessagesAfterUpdate)
+		def vpnConnectionStatusChangeMessages = bobMessagesAfterUpdate.responseData._embedded?."yona:messages".findAll{ it."@type" == "BuddyVpnConnectionStatusChangeMessage"}
+
+		vpnConnectionStatusChangeMessages.size() == 1
+		vpnConnectionStatusChangeMessages[0]._links.self != null
+		vpnConnectionStatusChangeMessages[0]._links."yona:user".href == bob.buddies[0].user.url
```

Comment:
Maybe a link to the device would be helpful too?

Reply:
It makes sense doing so, but isn't trivial and won't make it consistent. See [YD-556](https://jira.yona.nu/browse/YD-556).